meta-lmp-base/classes/uboot-fitimage.bbclass

```diff
@@ -25,6 +25,7 @@ ATF_BINARY ?= "arm-trusted-firmware.bin"
 ATF_SUPPORT = "${@bb.utils.contains('EXTRA_IMAGEDEPENDS', 'virtual/trusted-firmware-a', 'true', 'false', d)}"
 
 do_compile[depends] += " virtual/optee-os:do_deploy"
+do_compile[depends] += " u-boot-default-script:do_deploy"
```

Comment:
From the build perspective I think we should make this optional, kind of as done for ATF, since then we can enable/disable this support based on how the board is configured (e.g. allow u-boot.itb with normal boot script).

Like:

```
BOOTSCR_SUPPORT = "${@bb.utils.contains('PREFERRED_PROVIDER_u-boot-default-script', 'u-boot-ostree-scr-fit', 'true', 'false', d)}"
do_compile[depends] += " ${@'u-boot-default-script:do_deploy' if d.getVar('BOOTSCR_SUPPORT') == 'true' else ''}"

```
And then make the fit block conditional.

Reply:
I think we can store pre-load address of boot.scr instead of just true/false

Reply:
Sure, as long it can also be enabled/disabled based on the default u-boot-default-script provider, we should be fine.